src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -135,7 +142,30 @@ private[impl] class InstanceTrackerActor(
         val op = InstanceOpProcessor.Operation(deadline, sender(), instanceId, instanceUpdateOp)
         updaterRef.forward(InstanceUpdateActor.ProcessInstanceOp(op))
 
-      case msg @ InstanceTrackerActor.StateChanged(ack) =>
+      case InstanceTrackerActor.StateChanged(ack) =>
+        import context.dispatcher
+
+        ack.effect match {
+          case InstanceUpdateEffect.Update(instance, _, _) =>
+            updateRepository(() => repository.store(instance), ack)
+              .pipeTo(self)(sender())
+
+          case InstanceUpdateEffect.Expunge(instance, _) =>
+            logger.debug(s"Received expunge for ${instance.instanceId}")
+            updateRepository(() => repository.delete(instance.instanceId), ack)
+              .pipeTo(self)(sender())
+
+          case _ =>
+            Future.successful(RepositoryStateUpdated(ack)) // forward only
+              .pipeTo(self)(sender())
+        }
+
+      case Status.Failure(e) =>
+        // pipeTo self might cause failed future to be propagated as Status.Failure
+        // this should happen only for fatal exceptions
+        throw e
```

Comment:
You definitely want some logging here (on error level)!

Reply:
yep, I already fixed that locally in the morning :) thanks